### DISCUSSION TITLE
Lines rendering & camera

### DISCUSSION THREAD

**Discussion by ComradeMashkov** [pelicanmapping/rocky#43] [Q&A / Support] [answered]
Hey, I have 2 issues about these things:

1. Can I choose the main camera orientation and view on the start? For example, can I set camera initial lat, lon and alt? If I can, how can I do so?
2. I'm currently rendering lines by pressing the buttons on the GUI. My lines have small length and I prefer to render them only on certain zoom level. As you can see on the screenshot, lines are being rendered even on the far distance (min zoom level) and I want them to be shown after some zooming. Can I do this? Thanks for your reply!

<details>

<summary>Screenshot</summary>

<img width="960" alt="image" src="https://github.com/pelicanmapping/rocky/assets/117441107/32068ea8-e412-4dca-9fff-990302ba43e9">

</details>

**Comment by gwaldron** [comment, marked as answer]
Hello!

To set the viewpoint on startup, you need to call Application::realize() manually, set a viewpoint, and then start the frame loop.(I just made realize() public in the latest push so you will need to update.) Example:

```c++

    app.realize();

    auto first_view = app.displayManager->windowsAndViews.begin()->second.front();
    if (first_view)
    {
        auto manip = rocky::MapManipulator::get(first_view);
        if (manip)
        {
            rocky::Viewpoint vp;
            vp.pitch = -90.0;
            vp.point = rocky::GeoPoint(rocky::SRS::WGS84, 45, 45, 0);
            vp.range = 1000000.0;
            manip->setViewpoint(vp);
        }
    }
    app.run();
```

**Reply by GaussGun3000** [reply]
Using latest commit today, this example does not work properly, camera is always looking into outer space being pretty far from earth, no matter what pitch and range I set

**Reply by gwaldron** [reply]
I dropped this code into rocky_demo.cpp, and it appears to works correctly.
Can you please do the same and post a screenshot?

**Reply by GaussGun3000** [reply]
Black screen initially: https://snipboard.io/L4TGaN.jpg

The globe is still here, just need to pan a bit to find it: https://snipboard.io/M3PwJa.jpg

**Comment by gwaldron** [comment]
Configuring lines (and other primitives) to use LOD or screen-space error is not integrated yet, so in the meantime you'll need to manage this yourself by setting the `visible` flag on your components as necessary.

(The plan is to support SSE on components in general, and possibly to provide a hook whereby you can insert your own custom node processing (like making a parent LOD node) when components get created.)
